### ISSUE TITLE
No module named version_tools

### ISSUE THREAD

**Issue by invisibleroads** [pyamg/pyamg#192]
The build was working as of yesterday, but it looks like the new version on PyPI is missing a few files.

```
  Using cached pyamg-3.2.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-Bi6Pn1/pyamg/setup.py", line 22, in <module>
        import version_tools
    ImportError: No module named version_tools

```

**Comment by lukeolson** [comment]
Thanks.  Working on a fix -- I'm seeing this too.  There's an issue with the source distribution.

**Comment by invisibleroads** [comment]
You're fast!

Well, I beat you to it. I added a MANIFEST.in in #193

**Comment by lukeolson** [comment]
Yeah I was just adding a similar MANIFEST.  Does this fix it for you?  I'm still getting errors, now on version:

Traceback (most recent call last):
  File "setup.py", line 32, in <module>
    version_tools.set_version_info(version, isreleased)
  File "/Users/lukeo/tmp3/pyamg/dist/pyamg-3.2.0.dev0+4462794/version_tools.py", line 47, in set_version_info
    raise ImportError("Unable to import git_revision. Try removing "
ImportError: Unable to import git_revision. Try removing pyamg/version.py and the build directory before building.

**Comment by lukeolson** [comment]
There appear to be two issues

1. The files are not included in the source distribution correctly.  Fixing this is relatively easy through `MANIFEST.in` but it's frustrating that setuptools does not handle this more easily.
2. The source distribution tries to `from pyamg.version import ...` which fails.  This is trickier.  Version information needs to be at build time to pick up on git information.  The approach I used before was inherited from numpy and worked, but it was cluncky ...but it worked.  I'll try this approach again unless there are better ideas...

**Comment by invisibleroads** [comment]
Oh whoops you're right.

It looks like `version_tools` is trying to automatically identify the version of the package. Wouldn't it be simpler to just set the version in `setup.py` manually?

The problem is that `version_tools` is trying to import `pyamg/version.py`, but in order to import `pyamg`, it needs to import `pyamg/__init__.py` which imports modules that need `_amg_core`. However, `_amg_core` hasn't been generated yet by `setup.py`.

```
    elif os.path.exists('pyamg/version.py'):
        # must be a source distribution, use existing version file
        try:
            from pyamg.version import git_revision as GIT_REVISION
            from pyamg.version import git_branch as GIT_BRANCH
        except ImportError:
            raise ImportError("Unable to import git_revision. Try removing "
                              "pyamg/version.py and the build directory "
                              "before building.")

```

I think we're trying to automate too much and making the `setup` code too complex.

I recommend just setting the version manually in both `setup.py` and `pyamg.__version__` and forgetting about the git revision stuff.

**Comment by lukeolson** [comment]
Ok, fixed in 66ddb4242c964a50ee6a6f839cedffbb2e6cc957  (I should have squashed this but this is the relevant commit).  Summary: `MANIFEST.in` is added and the import is fixed using `imp`

**Comment by invisibleroads** [comment]
Great! We'll restore this package in our build.
https://github.com/crosscompute/crosscompute-environments-ansible
